src/snowflake/cli/api/project/schemas/project_definition.py

```diff
@@ -102,23 +104,53 @@ def _convert_env(
         return ProjectEnvironment(default_env=(env or {}), override_env={})
 
 
-class ProjectDefinition(_DefinitionV11):
-    def __init__(self, **kwargs):
-        super().__init__(**kwargs)
-        self._validate(kwargs)
+class DefinitionV20(_ProjectDefinitionBase):
+    entities: Dict = Field(
+        title="Entity definitions.",
+    )
+
+    env: Union[Dict[str, str], ProjectEnvironment, None] = Field(
+        title="Environment specification for this project.",
+        default=None,
+        validation_alias="env",
+        union_mode="smart",
+    )
+
+    @field_validator("env")
+    @classmethod
+    def _convert_env(
+        cls, env: Union[Dict, ProjectEnvironment, None]
+    ) -> ProjectEnvironment:
+        if isinstance(env, ProjectEnvironment):
+            return env
+        return ProjectEnvironment(default_env=(env or {}), override_env={})
+
+    @field_validator("entities")
+    @classmethod
+    def validate_entities(cls, entities: Dict) -> Dict:
+        # TODO Add entities validation logic
+        return entities
+
+
+def build_project_definition(**data):
+    """
+    Returns a ProjectDefinition instance with a version matching the provided definition_version value
+    """
+    if not isinstance(data, dict):
+        return
+    version = data.get("definition_version")
+    version_model = get_version_map().get(str(version))
+    if not version or not version_model:
+        # Raises a SchemaValidationError
+        _ProjectDefinitionBase(**data)
+    return version_model(**data)
+
 
-    @staticmethod
-    def _validate(data: Any):
-        if not isinstance(data, dict):
-            return
-        if version := str(data.get("definition_version")):
-            version_model = _version_map.get(version)
-            if not version_model:
-                raise ValueError(
-                    f"Unknown schema version: {version}. Supported version: {_supported_version}"
-                )
-            version_model(**data)
+ProjectDefinition = Union[DefinitionV10, DefinitionV11, DefinitionV20]
 
 
-_version_map = {"1": _DefinitionV10, "1.1": _DefinitionV11}
-_supported_version = tuple(_version_map.keys())
+def get_version_map():
```

Comment:
nit: _get_version_map? It was private before. Unless I missed something, it's still only used here.

Reply:
Ah, just saw the comments above, ignore this, it was intentional.